src/rule_definition_tools.jl

```diff
@@ -356,7 +357,7 @@ function tuple_expression(primal_sig_parts)
     else
         num_primal_inputs = length(primal_sig_parts) - 1 # - vararg
         length_expr = :($num_primal_inputs + length($(esc(_unconstrain(primal_sig_parts[end])))))
-        Expr(:call, :ntuple, Expr(:(->), :_, DoesNotExist()), length_expr)
+        Expr(:call, :ntuple, Expr(:(->), :($(esc(:_))), DoesNotExist()), length_expr)
```

Comment:
We should not rely on this as a fix, since this behavior will likely change soon.

Reply:
What do you suggest? Using `::Any` instead? Or just `:_`? I also noticed that if `esc` is removed both in the pullback function signature and this anonymous function, then the macro hygiene will replace both occurrences of `_` with the same variable name (something like `var"#249#_").

Reply:
Ah, in that case I'd go with `::Any` or just something like `_tmp`.